src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,59 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_AGE = 18;
+    private static final int MINIMAL_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User is null!");
+        }
+        String login = user.getLogin();
+        if (storageDao.get(login) != null) {
+            throw new RegistrationException("User with given login \""
+                    + login + "\" already exists");
+        }
+        validateLogin(login);
+        validatePassword(user.getPassword());
+        validateAge(user.getAge());
```

Comment:
caller should not be bothered with extra details - you can pass user to each method and it is a responsibility of that method to validate what necessary
```suggestion
        validateLogin(user);
        validatePassword(user);
        validateAge(user);
```